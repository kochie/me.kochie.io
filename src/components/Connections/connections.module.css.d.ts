export const connections: string
export const verticalMiddle: string
export const hoverActive: string
export const gitlab: string
export const medium: string
export const github: string
export const instagram: string
export const twitter: string
export const linkedin: string
export const mastodon: string
export const grow: string
