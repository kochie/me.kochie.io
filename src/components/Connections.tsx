import * as React from "react";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faLinkedin,
  faTwitter,
  faInstagram,
  faGithub,
  faMediumM,
  faGitlab
} from "@fortawesome/free-brands-svg-icons";

import "../styles/Connections.scss";

const connections = [
  {
    icon: faLinkedin,
    href: "https://linkedin.com/in/rkkochie",
    name: "rkkochie",
    color: "linkedin"
  },
  {
    icon: faTwitter,
    href: "https://twitter.com/rkkochie",
    name: "rkkochie",
    color: "twitter"
  },
  {
    icon: faInstagram,
    href: "https://instagram.com/rkkochie",
    name: "rkkochie",
    color: "instagram"
  },
  {
    icon: faGithub,
    href: "https://github.com/kochie",
    name: "kochie",
    color: "github"
  },
  {
    icon: faMediumM,
    href: "https://medium.com/@kochie",
    name: "kochie",
    color: "medium"
  },
  {
    icon: faGitlab,
    href: "https://gitlab.com/kochie",
    name: "kochie",
    color: "gitlab"
  }
];

export default class Connections extends React.Component {
  render() {
    return (
      <div className="connections">
        {connections.map(connection => (
          <div className="grow">
            <a href={connection.href} className={`${connection.color} grow`}>
              <FontAwesomeIcon
                icon={connection.icon}
                //color={connection.color}
                size="2x"
                fixedWidth
                className="vertical-middle"
              />
              {/* <span>{connection.name}</span> */}
            </a>
          </div>
        ))}
      </div>
    );
  }
}
