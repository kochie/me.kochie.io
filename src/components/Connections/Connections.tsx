import * as React from "react";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faLinkedin,
  faTwitter,
  faInstagram,
  faGithub,
  faMediumM,
  faGitlab,
  faMastodon
} from "@fortawesome/free-brands-svg-icons";

import Head from "next/head";

const connections = [
  {
    icon: faLinkedin,
    href: "https://linkedin.com/in/rkkochie",
    name: "rkkochie",
    color: "linkedin"
  },
  {
    icon: faTwitter,
    href: "https://twitter.com/kochie",
    name: "rkkochie",
    color: "twitter"
  },
  {
    icon: faInstagram,
    href: "https://instagram.com/rkkochie",
    name: "rkkochie",
    color: "instagram"
  },
  {
    icon: faGithub,
    href: "https://github.com/kochie",
    name: "kochie",
    color: "github"
  },
  {
    icon: faMediumM,
    href: "https://medium.com/@kochie",
    name: "kochie",
    color: "medium"
  },
  {
    icon: faGitlab,
    href: "https://gitlab.com/kochie",
    name: "kochie",
    color: "gitlab"
  },
  {
    icon: faMastodon,
    href: "https://melb.social/@kochie",
    name: "kochie",
    color: "mastodon"
  }
];

export default class Connections extends React.Component {
  render() {
    return (
      <>
        <Head>
          <link rel="stylesheet" href="/static/styles/connections.css" />
        </Head>
        <div className="connections">
          {connections.map(connection => (
            <div className="grow" key={connection.color}>
              <a href={connection.href} className={`${connection.color} grow`}>
                <FontAwesomeIcon
                  icon={connection.icon}
                  size="2x"
                  fixedWidth
                  className="vertical-middle"
                />
              </a>
            </div>
          ))}
        </div>
      </>
    );
  }
}
