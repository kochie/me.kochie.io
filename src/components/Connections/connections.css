.connections {
  color: wheat;
  font-family: "Roboto Mono", monospace;
  padding-top: 50px;
  display: flex;
  flex-direction: column;
  align-items: center;
  /* gap: 10px; */
}

.connections > ul {
  list-style: none;
  padding: 0;
  margin: 0;
  text-align: left;
}

.connections span {
  padding-left: 5px;
  color: wheat;
}

.connections a {
  color: wheat;
}

.vertical-middle {
  vertical-align: middle;
}

.hover-active:hover {
  color: green;
}

.gitlab:hover {
  color: #db3b21;
}

.medium:hover {
  color: #000000;
}

.github:hover {
  color: #333333;
}

.instagram:hover {
  color: #833ab4;
}

.twitter:hover {
  color: #1da1f2;
}

.linkedin:hover {
  color: #0077b5;
}

.mastodon:hover {
  color: #008ada;
}

.grow {
  transition: all 0.2s ease-in-out;
  width: 35px;
  padding: 5px 0;
}
.grow:hover {
  transform: scale(1.1);
}

@media (--tablet-size) {
  .connections {
    flex-direction: row;
    justify-content: center;
  }

  .grow {
    padding: 0 10px;
  }
}

@media (--mobile-size) {
  .connections {
    flex-wrap: wrap;
    max-width: 300px;
  }
  .grow {
    padding: 10px 10px;
  }
}
