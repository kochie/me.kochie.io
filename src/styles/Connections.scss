@import "constants";

.connections {
  color: wheat;
  font-family: "Roboto Mono", monospace;
  padding-top: 50px;
  display: flex;
  flex-direction: column;
  @media only screen and (max-width: $mobile-width) and (max-height: $mobile-height) and (min-resolution: $mobile-resolution),
    only screen and (max-width: $mobile-height) and (max-height: $mobile-width) and (min-resolution: $mobile-resolution) {
      flex-direction: row;
      justify-content: center;
  }
}

.connections > ul {
  list-style: none;
  padding: 0;
  margin: 0;
  text-align: left;
}

.connections span {
  padding-left: 5px;
  color: wheat;
}

.connections a {
  color: wheat;
}

.vertical-middle {
  vertical-align: middle;
  // padding: 3px;
}

.hover-active:hover {
  color: green;
}

.gitlab:hover {
  color: #db3b21;
}

.medium:hover {
  color: #000000;
}

.github:hover {
  color: #333333;
}

.instagram:hover {
  color: #833ab4;
}

.twitter:hover {
  color: #1da1f2;
}

.linkedin:hover {
  color: #0077b5;
}

.grow {
  transition: all 0.2s ease-in-out;
  padding: 5px;
  @media only screen and (max-width: $mobile-width) and (max-height: $mobile-height) and (min-resolution: $mobile-resolution),
    only screen and (max-width: $mobile-height) and (max-height: $mobile-width) and (min-resolution: $mobile-resolution) {
      padding: 0px 1px;
  }
}
.grow:hover {
  transform: scale(1.1);
  padding: 5px;
  @media only screen and (max-width: $mobile-width) and (max-height: $mobile-height) and (min-resolution: $mobile-resolution),
    only screen and (max-width: $mobile-height) and (max-height: $mobile-width) and (min-resolution: $mobile-resolution) {
      padding: 0px 1px;
  }
}
