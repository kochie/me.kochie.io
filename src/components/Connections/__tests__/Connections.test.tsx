import renderer from 'react-test-renderer'
import React from 'react'

import Connections from '../Connections'

import { fab } from '@fortawesome/free-brands-svg-icons'
import { library } from '@fortawesome/fontawesome-svg-core'
library.add(fab)

it('connections renders correctly', () => {
  const tree = renderer.create(<Connections />).toJSON()
  expect(tree).toMatchSnapshot()
})
