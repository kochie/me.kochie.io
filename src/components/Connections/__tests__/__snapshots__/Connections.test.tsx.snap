// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connections renders correctly 1`] = `
<div
  className="text-wheat pt-12 flex xl:flex-col flex-row items-center gap-3"
>
  <div
    className="w-9 transform-gpu transition duration-200 ease-in-out hover:scale-110"
  >
    <a
      aria-label="linkedin"
      className="hover:text-linkedin"
      href="https://linkedin.com/in/rkkochie"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-linkedin fa-w-14 fa-2x align-middle"
        data-icon="linkedin"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </div>
  <div
    className="w-9 transform-gpu transition duration-200 ease-in-out hover:scale-110"
  >
    <a
      aria-label="twitter"
      className="hover:text-twitter"
      href="https://twitter.com/kochie"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-twitter fa-w-16 fa-2x align-middle"
        data-icon="twitter"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </div>
  <div
    className="w-9 transform-gpu transition duration-200 ease-in-out hover:scale-110"
  >
    <a
      aria-label="instagram"
      className="hover:text-instagram"
      href="https://instagram.com/rkkochie"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-instagram fa-w-14 fa-2x align-middle"
        data-icon="instagram"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </div>
  <div
    className="w-9 transform-gpu transition duration-200 ease-in-out hover:scale-110"
  >
    <a
      aria-label="github"
      className="hover:text-github"
      href="https://github.com/kochie"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-github fa-w-16 fa-2x align-middle"
        data-icon="github"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 496 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </div>
  <div
    className="w-9 transform-gpu transition duration-200 ease-in-out hover:scale-110"
  >
    <a
      aria-label="medium-m"
      className="hover:text-medium"
      href="https://medium.com/@kochie"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-medium-m fa-w-16 fa-2x align-middle"
        data-icon="medium-m"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M71.5 142.3c.6-5.9-1.7-11.8-6.1-15.8L20.3 72.1V64h140.2l108.4 237.7L364.2 64h133.7v8.1l-38.6 37c-3.3 2.5-5 6.7-4.3 10.8v272c-.7 4.1 1 8.3 4.3 10.8l37.7 37v8.1H307.3v-8.1l39.1-37.9c3.8-3.8 3.8-5 3.8-10.8V171.2L241.5 447.1h-14.7L100.4 171.2v184.9c-1.1 7.8 1.5 15.6 7 21.2l50.8 61.6v8.1h-144v-8L65 377.3c5.4-5.6 7.9-13.5 6.5-21.2V142.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </div>
  <div
    className="w-9 transform-gpu transition duration-200 ease-in-out hover:scale-110"
  >
    <a
      aria-label="gitlab"
      className="hover:text-gitlab"
      href="https://gitlab.com/kochie"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-gitlab fa-w-16 fa-2x align-middle"
        data-icon="gitlab"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M105.2 24.9c-3.1-8.9-15.7-8.9-18.9 0L29.8 199.7h132c-.1 0-56.6-174.8-56.6-174.8zM.9 287.7c-2.6 8 .3 16.9 7.1 22l247.9 184-226.2-294zm160.8-88l94.3 294 94.3-294zm349.4 88l-28.8-88-226.3 294 247.9-184c6.9-5.1 9.7-14 7.2-22zM425.7 24.9c-3.1-8.9-15.7-8.9-18.9 0l-56.6 174.8h132z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </div>
  <div
    className="w-9 transform-gpu transition duration-200 ease-in-out hover:scale-110"
  >
    <a
      aria-label="mastodon"
      className="hover:text-mastodon"
      href="https://melb.social/@kochie"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-mastodon fa-w-14 fa-2x align-middle"
        data-icon="mastodon"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M433 179.11c0-97.2-63.71-125.7-63.71-125.7-62.52-28.7-228.56-28.4-290.48 0 0 0-63.72 28.5-63.72 125.7 0 115.7-6.6 259.4 105.63 289.1 40.51 10.7 75.32 13 103.33 11.4 50.81-2.8 79.32-18.1 79.32-18.1l-1.7-36.9s-36.31 11.4-77.12 10.1c-40.41-1.4-83-4.4-89.63-54a102.54 102.54 0 0 1-.9-13.9c85.63 20.9 158.65 9.1 178.75 6.7 56.12-6.7 105-41.3 111.23-72.9 9.8-49.8 9-121.5 9-121.5zm-75.12 125.2h-46.63v-114.2c0-49.7-64-51.6-64 6.9v62.5h-46.33V197c0-58.5-64-56.6-64-6.9v114.2H90.19c0-122.1-5.2-147.9 18.41-175 25.9-28.9 79.82-30.8 103.83 6.1l11.6 19.5 11.6-19.5c24.11-37.1 78.12-34.8 103.83-6.1 23.71 27.3 18.4 53 18.4 175z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </div>
</div>
`;
